client/asset/dcr/dcr.go

```diff
@@ -759,6 +784,30 @@ func (dcr *ExchangeWallet) feeRate(confTarget uint64) (uint64, error) {
 	return 1 + uint64(atomsPerKB)/1000, nil // dcrPerKB * 1e8 / 1e3
 }
 
+// externalFeeEstimator gets the fee rate from the external API
+func externalFeeEstimator(nb uint64) (float64, error) {
+	url := externalApiUrl + "/utils/estimatefee?nbBlocks=" + strconv.Itoa(int(nb))
+	r, err := http.Get(url)
+	if err != nil {
+		return 0, err
+	}
+	body, err := ioutil.ReadAll(r.Body)
+	r.Body.Close()
+	if err != nil {
+		return 0, err
+	}
+	c := make(map[uint64]json.RawMessage)
+	err = json.Unmarshal(body, &c)
```

Comment:
`resp.Body` is a `io.Reader`, so could just use `json.NewDecoder(resp.Body).Decode` directly instead of `ReadAll` (still with a read limit).